testsuite/MDAnalysisTests/coordinates/test_imd.py

```diff
@@ -0,0 +1,418 @@
+
+    @staticmethod
+    @pytest.fixture()
+    def transformed(ref):
+        # This will start the test IMD Server, waiting for a connection
+        # to then send handshake & first frame
+        ref.server.handshake_sequence("localhost")
+        # This will connect to the test IMD Server and read the first frame
+        transformed = ref.reader(
+            f"{ref.trajectory}:{ref.server.port}",
+            n_atoms=ref.n_atoms,
+            buffer_size=1 * 1024 * 1024,
+        )
+        # Send the rest of the frames- small enough to all fit in socket itself
+        ref.server.send_frames(1, 5)
+        transformed.add_transformations(
+            translate([1, 1, 1]), translate([0, 0, 0.33])
+        )
+        return transformed
+
+    def test_n_frames(self, ref, reader):
+        pytest.skip("`n_frames` is unknown for IMDReader")
+
+    def test_first_frame(self, ref, reader):
+        # don't rewind here as in inherited base test
+        assert_timestep_almost_equal(
+            reader.ts, ref.first_frame, decimal=ref.prec
+        )
+
+    def test_get_writer_1(self, ref, reader, tmpdir):
+        pytest.skip("No Writer for IMDReader")
+
+    def test_get_writer_2(self, ref, reader, tmpdir):
+        pytest.skip("No Writer for IMDReader")
+
+    def test_total_time(self, ref, reader):
+        pytest.skip("`total_time` is unknown for IMDReader")
+
+    def test_changing_dimensions(self, ref, reader):
+        pytest.skip("IMDReader cannot be rewound")
+
+    def test_iter(self, ref, reader):
+        for i, ts in enumerate(reader):
+            assert_timestep_almost_equal(ts, ref.iter_ts(i), decimal=ref.prec)
+
+    def test_first_dimensions(self, ref, reader):
+        # don't rewind here as in inherited base test
+        if ref.dimensions is None:
+            assert reader.ts.dimensions is None
+        else:
+            assert_allclose(
+                reader.ts.dimensions,
+                ref.dimensions,
+                rtol=0,
+                atol=1.5 * 10 ** (-ref.prec),
+            )
+
+    def test_transformed(self, ref, transformed):
+        # see transformed fixture
+        ref_trans = ref.first_frame.positions + 1
+        ref_trans[:, 2] += 0.33
+        assert_allclose(transformed.ts.positions, ref_trans)
+
+    def test_volume(self, ref, reader):
+        # don't rewind here as in inherited base test
+        vol = reader.ts.volume
+        # Here we can only be sure about the numbers upto the decimal point due
+        # to limited floating point precision.
+        assert_allclose(vol, ref.volume, rtol=0, atol=1.5e0)
+
+    def test_reload_auxiliaries_from_description(self, ref, reader):
+        pytest.skip("Cannot create two IMDReaders on the same stream")
+
+    def test_stop_iter(self, reader):
+        pytest.skip("IMDReader cannot be rewound")
+
+    def test_iter_rewinds(self, reader):
+        pytest.skip("IMDReader cannot be rewound")
+
+    def test_timeseries_shape(self, reader):
```

Comment:
One test should check that timeseries raises RuntimeError.

The others can be skips.

Reply:
Ok, tested below in `test_timeseries_raises`